src/app/base/components/PrefixedIpInput/PrefixedIpInput.tsx

```diff
@@ -18,46 +21,76 @@ type Props = Omit<
 };
 
 const PrefixedIpInput = ({ cidr, name, ...props }: Props) => {
+  const [networkAddress] = cidr.split("/");
+  const ipv6Prefix = networkAddress.substring(
+    0,
+    networkAddress.lastIndexOf(":")
+  );
+  const subnetIsIpv4 = isIPv4(networkAddress);
+
   const [startIp, endIp] = getIpRangeFromCidr(cidr);
   const [immutable, editable] = getImmutableAndEditableOctets(startIp, endIp);
 
   const formikProps = useFormikContext();
 
+  const getPlaceholderText = () => {
+    if (subnetIsIpv4) {
+      return editable;
+    } else {
+      // 7 is the maximum number of colons in an IPv6 address
+      const placeholderColons = 7 - (ipv6Prefix.match(/:/g) || []).length;
+      return `${"0000:".repeat(placeholderColons)}0000`;
+    }
+  };
+
   const getMaxLength = () => {
-    const immutableOctetsLength = immutable.split(".").length;
+    if (subnetIsIpv4) {
+      const immutableOctetsLength = immutable.split(".").length;
+
+      if (immutableOctetsLength === 3) {
+        return 3; // 3 digits, no dots
+      } else if (immutableOctetsLength === 2) {
+        return 7; // 6 digits, 1 dot
+      } else if (immutableOctetsLength === 1) {
+        return 11; // 9 digits, 2 dots
+      } else {
+        return 15; // 12 digits, 3 dots
+      }
+    } else {
+      return getPlaceholderText().length;
+    }
+  };
```

Comment:
I'd suggest refactoring to simplify the logic.

```tsx
  const getIPv6Placeholder = () => {
    // 7 is the maximum number of colons in an IPv6 address
    const placeholderColons = 7 - (ipv6Prefix.match(/:/g) || []).length;
    return `${"0000:".repeat(placeholderColons)}0000`;
  };

  const getPlaceholderText = () =>
    subnetIsIpv4 ? editable : getIPv6Placeholder();

  const getIPv4MaxLength = () => {
    const immutableOctetsLength = immutable.split(".").length;
    const lengths = [15, 11, 7, 3]; // Corresponding to 0-3 immutable octets
    return lengths[immutableOctetsLength];
  };

  const getMaxLength = () =>
    subnetIsIpv4 ? getIPv4MaxLength() : getIPv6Placeholder().length;
```

- Clearly separates the logic for IPv4 and IPv6, making it easier to follow.
- Uses an array for IPv4 lengths, which simplifies the logic.

Using data structures for lookup is _generally a good idea_, next time when you're dealing with switch-like statements, consider using this pattern.